commit 910e43a0b0a25232db7fa7942b2efad4e60cb904
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Fri Mar 10 00:04:06 2017 +0100

    Backport BUILD_TOOLS switch from KF5Akonadi
    
    This makes it possible to disable colliding build environment files.

diff --git a/akonadi/CMakeLists.txt b/akonadi/CMakeLists.txt
index 4a4245408..4dfbe2ab1 100644
--- a/akonadi/CMakeLists.txt
+++ b/akonadi/CMakeLists.txt
@@ -7,11 +7,22 @@ if(CMAKE_COMPILE_GCOV)
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
 endif()
 
-if (KDE4_BUILD_TESTS)
-  add_subdirectory( tests )
-else()
+option(BUILD_TOOLS "Build and install tools for development and testing purposes." TRUE)
+
+if(KDE4_BUILD_TESTS)
+  set(BUILD_TOOLS TRUE)
+endif()
+
+if (BUILD_TOOLS)
+# add testrunner (application for managing a self-contained test environment)
   add_subdirectory( tests/testrunner )
   add_subdirectory( tests/testresource )
+  add_subdirectory( tests/testsearchplugin )
+  add_subdirectory( tests/benchmarker )
+endif()
+
+if (KDE4_BUILD_TESTS)
+  add_subdirectory( tests )
 endif()
 
 add_definitions( ${AKONADI_DEFINITIONS} )
diff --git a/akonadi/tests/CMakeLists.txt b/akonadi/tests/CMakeLists.txt
index ffd16ff22..1675e354d 100644
--- a/akonadi/tests/CMakeLists.txt
+++ b/akonadi/tests/CMakeLists.txt
@@ -26,16 +26,7 @@ include_directories(
   ${CMAKE_BINARY_DIR}/kmime
 )
 
-# add testrunner (application for managing a self-contained test
-# environment)
-
 add_subdirectory(etm_test_app)
-add_subdirectory(testrunner)
-add_subdirectory(testresource)
-add_subdirectory(testsearchplugin)
-
-# add benchmarker
-add_subdirectory(benchmarker)
 
 # convenience macro to add akonadi demo application
 macro(add_akonadi_demo _source)
diff --git a/akonadi/xml/CMakeLists.txt b/akonadi/xml/CMakeLists.txt
index 18ab0e1bd..dc85df63d 100644
--- a/akonadi/xml/CMakeLists.txt
+++ b/akonadi/xml/CMakeLists.txt
@@ -2,7 +2,9 @@ if ( LIBXML2_FOUND )
   add_definitions( -DHAVE_LIBXML2 )
 endif ()
 
-add_subdirectory( tests )
+if (KDE4_BUILD_TESTS)
+  add_subdirectory( tests )
+endif()
 
 include_directories(
     ${KDE4_INCLUDES}
@@ -41,11 +43,13 @@ install(TARGETS akonadi-xml EXPORT kdepimlibsLibraryTargets ${INSTALL_TARGETS_DE
 install(FILES ${akonadixml_HDRS} DESTINATION ${INCLUDE_INSTALL_DIR}/akonadi/xml COMPONENT Devel)
 install(FILES akonadi-xml.xsd DESTINATION ${DATA_INSTALL_DIR}/akonadi/)
 
-kde4_add_executable(akonadi2xml akonadi2xml.cpp)
-target_link_libraries(akonadi2xml
-   akonadi-xml
-   ${KDE4_KDEUI_LIBS}
-   akonadi-kde
-)
+if(BUILD_TOOLS)
+  kde4_add_executable(akonadi2xml akonadi2xml.cpp)
+  target_link_libraries(akonadi2xml
+    akonadi-xml
+    ${KDE4_KDEUI_LIBS}
+    akonadi-kde
+  )
 
-install(TARGETS akonadi2xml ${INSTALL_TARGETS_DEFAULT_ARGS})
+  install(TARGETS akonadi2xml ${INSTALL_TARGETS_DEFAULT_ARGS})
+endif()
